### PR TITLE
Lollipop machine and fruity treats V2

### PR DESCRIPTION
**Pull request by Hypertripp**
## About the PR 
Remember when you were small and after a visit to the doctor you got a lollipop? Well NT has decided that you all are practically children so now the doctor can menace the dentist and prescribe sugary treats to any staffie they see!
And it seems the candy salesman found one to, now he doesn't have to worry about running out.

## Why's this needed? 
It gives the doctor a fun new rp interaction of giving out a lollipop when they are done with a patient
And if I made a candy maker and not just give it to candy salesman it would be a crime

## Changelog

```changelog(u)Hypertripp
(*)Added fruit lollipops and a dispenser  to vend them.
```

### CODE REVIEW

File: code/modules/food_and_drink/candy.dm [hunk 1]
```diff
@@ -331,6 +331,20 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/candy/jellybean)
 				for (var/i=5, i>0, i--)
 					src.reagents.add_reagent(pick(src.flavors), 1)
 
+/obj/item/reagent_containers/food/snacks/lollipop/fruit
+	icon_state = "lpop-"
```
[the remark this answers is not in the document]
Comment (mordent-goonstation) [reply code/modules/food_and_drink/candy.dm:335]: Counterpoint: you should understand code you copy/paste.

File: code/datums/jobs.dm [hunk 2]
```diff
@@ -678,7 +678,7 @@ ABSTRACT_TYPE(/datum/job/research)
 	slot_eyes = list(/obj/item/clothing/glasses/healthgoggles/upgraded)
 	slot_poc1 = list(/obj/item/device/pda2/medical)
 	slot_poc2 = list(/obj/item/paper/book/from_file/pocketguide/medical)
-	items_in_backpack = list(/obj/item/crowbar, /obj/item/robodefibrillator) // cogwerks: giving medics a guaranteed air tank, stealing it from roboticists (those fucks)
+	items_in_backpack = list(/obj/item/crowbar, /obj/item/robodefibrillator, /obj/item/device/lollipopmaker) // cogwerks: giving medics a guaranteed air tank, stealing it from roboticists (those fucks)
```
[the remark this answers is not in the document]
Comment (mordent-goonstation) [reply code/datums/jobs.dm:681]: If it's not heavily limited in its usage, might be a fun thing to put in the medical breakrooms.

File: code/obj/item/device/lollipop_machine.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		if(ON_COOLDOWN(src, "vend_cooldown", 10 SECONDS))
+			user.show_text("It's still recharging, give it a moment. ", "red")
+
+		else
+			new /obj/item/reagent_containers/food/snacks/lollipop/fruit(get_turf(src))
+			playsound(src.loc, "sound/machines/click.ogg", 50, 1)
+			user.visible_message("<span class='notice'>[user] dispenses a lollipop.</span>")
+		return
+
+/obj/item/device/lollipopmaker/md
```
[the remark this answers is not in the document]
Comment (mordent-goonstation) [reply code/obj/item/device/lollipop_machine.dm:17]: Gotcha.

File: code/obj/item/device/lollipop_machine.dm [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "lollipop-h"
+
+	attack_self(var/mob/user as mob)
+
+		if(ON_COOLDOWN(src, "vend_cooldown", 10 SECONDS))
+			user.show_text("It's still recharging, give it a moment. ", "red")
+
+		else
+			new /obj/item/reagent_containers/food/snacks/lollipop/fruit(get_turf(src))
```
[the remark this answers is not in the document]
Comment (mordent-goonstation) [reply code/obj/item/device/lollipop_machine.dm:13]: Yeah, not end of the world. Infinite and zero-cost *anything* though is bound to be abused for something. Fuel for loafers, singularity top-up, etc. etc.

[the remark this answers is not in the document]
Comment (Hypertripp) [reply code/obj/item/device/lollipop_machine.dm:13]: the current plan is a 40 pu battery with a 2.5 recharge rate and a10 pu to use

File: code/modules/food_and_drink/candy.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hypertripp) [reply code/modules/food_and_drink/candy.dm:335]: ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (Hypertripp) [reply code/modules/food_and_drink/candy.dm:335]: best way to learn is hands on

File: code/modules/food_and_drink/candy.dm [hunk 5]
```diff
@@ -331,6 +331,20 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/candy/jellybean)
...
+
+	New()
+		..()
+		src.reagents.inert = true
+		SPAWN_DBG(0)
+			reagents.add_reagent("sugar",5)
+			if (src.icon_state == "lpop-")
+				src.icon_state = "lpop-[rand(1,6)]"
+			if (islist(src.flavors) && length(src.flavors))
+				src.reagents.add_reagent(pick(src.flavors), 5)
```
[the remark this answers is not in the document]
Comment (Zonespace27) [reply code/modules/food_and_drink/candy.dm:346]: > Yah, iv tried looking up and down but i haven't the faintest idea how to match the two, but its consistent with the medical pops. also if they did share a color the sugar would make them all a dull gray version of the fruit color, NT dosent invest in food dye

what you'd want to do is something like this:
```
var/flavornum = rand(1, 6)
src.icon_state = "lpop-[flavornum]"
switch(flavornum)
   if(1)
      src.reagents.add_reagent("some_reagent", 5)
   if(2)
      src.reagents.add_reagent("some_other_reagent", 5)
   etc.
```

[the remark this answers is not in the document]
Comment (Hypertripp) [reply code/modules/food_and_drink/candy.dm:346]: Thank you Zonespace

**Comment by Hypertripp** [comment]
> Does this have in-hand sprites?

they do now

**Pull request closed without merging**